twisted/internet/base.py

```diff
@@ -601,18 +602,33 @@ def sigInt(self, *args):
         """Handle a SIGINT interrupt.
         """
         log.msg("Received SIGINT, shutting down.")
+        try:
+            import signal
```

Comment:
@adiroiban I have no idea how this code base works at all.  I'm working on a project, which uses a project that uses twisted.internet.  And I needed to be able to know when it died from a signal versus when it exited on its own.

Is there some other place this code should have gone?

Otherwise, what comment specifically are you wanting here?  I don't have the bigger picture context to understand what you're asking.